apps/aws-notebook/samples/ecs/fargate/vpn/main.tf

```diff
@@ -0,0 +1,252 @@
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 4.0"
+    }
+  }
+}
+
+# Configure the AWS Provider
+provider "aws" {
+}
+
+data "external" "tls" {
+  program = ["./prepare_openvpn_cert.sh"]
+}
+
+data "external" "vpn_client" {
+  program = ["./build_openvpn_config.sh"]
+
+  query = {
+    vpn_server_dns = aws_ec2_client_vpn_endpoint.dev-connection-ecs-fargate-vpn.dns_name
+  }
+}
+
+data "aws_iam_policy" "policy" {
+  name = "AmazonECSTaskExecutionRolePolicy"
+}
+
+resource "aws_iam_role" "role" {
+  name = "craftingDemoEcsTaskExecutionRole"
+
+  assume_role_policy = <<EOF
+    {
+      "Version": "2012-10-17",
+        "Statement": [
+        {
+          "Sid": "",
+          "Effect": "Allow",
+          "Principal": {
+            "Service": "ecs-tasks.amazonaws.com"
+          },
+          "Action": "sts:AssumeRole"
+        }
+        ]
+    }
+  EOF
+}
+
+resource "aws_iam_role_policy_attachment" "test-attach" {
+  role       = aws_iam_role.role.name
+  policy_arn = data.aws_iam_policy.policy.arn
+}
+
+data "aws_iam_role" "ecs_task_execution_role" {
+  #name = "ecsTaskExecutionRole"
```

Comment:
cleanup

Reply:
Why need to define a data resource here? We can directly use `aws_iam_role.role`